src/compiler/crystal/codegen/target.cr

```diff
@@ -32,6 +32,17 @@ class Crystal::Codegen::Target
     else
       # no need to tweak the architecture
     end
+
+    if linux? && environment_parts.size == 1
```

Comment:
Previous version looked IMO clearer:

```suggestion
    if linux? && !gnu? && !musl?
```

Reply:
This could lead to potentially incorrect behaviour when `environment` is something more complex and we're just appending `-gnu`. I don't think there are any practical issues with that.

The check for `!gnu? && !musl?` is a bit more expressive, though.

Reply:
It will catch triples that have libc that is neither gnu or musl: `"x86_64-linux-android"` => `"x86_64-unknown-linux-android-gnu"`

Reply:
`unknown` is one of the vendors that get the `-gnu` suffix however:

```crystal
require "compiler/requires"

target = Crystal::Codegen::Target.new("x86_64-linux-android")
p target.to_s # => "x86_64-unknown-linux-android-gnu"
```

(https://github.com/llvm/llvm-project/blob/main/clang/lib/Driver/ToolChains/Gnu.cpp#L2246)